projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -67,7 +67,12 @@ export class TableCdkDataSource implements DataSource<TableRow> {
     return this.rowsChange$.pipe(
       tap(rows => this.cacheRows(rows)),
       tap(rows => this.cacheFilterableValues(rows)),
-      tap(rows => this.loadingStateSubject.next({ loading$: of(rows), hide: rows.length > 0 }))
+      tap(rows => this.loadingStateSubject.next({ loading$: of(rows), hide: rows.length > 0 })),
```

Comment:
The changes in this file allow us to clear the data on error (previously, we were returning an empty observable (`[]` as an observable input) which the cdk would ignore causing the error icon to overlap with the previous data. Returning actual empty data at the old catch location isn't feasible since the no data check came later.